### PR TITLE
DAOS-15851 ci: Use CI_PR_REPOS parameter if provided

### PR DESCRIPTION
**Pull request by grom72**
Required for: https://github.com/daos-stack/go-build-toolchain/pull/25/commits/c8fde3edc5f3adee464db95228b67535f2d21031

Example build results:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14387/7/parameters/

https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-14387/7/pipeline-console/log?nodeId=189

```
...
[2024-05-20T19:29:27.522Z] PR_REPOS="go-build-toolchain@PR-25:9"                                  \
...
[2024-05-20T19:41:30.450Z] DEBUG util.py:448:  go-build-toolchain:PR-25:9                      554 kB/s |  58 kB     00:00    
...
```
It also add one memcheck suppresion related to Go lang.
### Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
This looks like what we are after, but let's not land it until we see it working as it's expected to using a calling project.  -1 is just until we can confirm that, to prevent an unintentional pre-mature landing.

**Comment by grom72** [comment]
@brianjmurrell please take a look at:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/ci-daos-stack-go-build-toolchain-PR-25-master/31/parameters/
and confirm that the solution works as you expect.

**Comment by brianjmurrell** [comment]
> @brianjmurrell please take a look at: [build.hpdd.intel.com/job/daos-stack/job/daos/job/ci-daos-stack-go-build-toolchain-PR-25-master/31/parameters](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/ci-daos-stack-go-build-toolchain-PR-25-master/31/parameters/) and confirm that the solution works as you expect.

Excellent work!  It does work as expected, for the most part.  As we can see in the [EL8 RPM build log](https://jenkins-1.wolf.hpdd.intel.com:9090/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/ci-daos-stack-go-build-toolchain-PR-25-master/runs/32/nodes/181/steps/486/log/?start=0):

```
[2024-05-22T15:07:55.555Z] DEBUG util.py:448:   golang                    x86_64  1.22.3-1.daos.el8                    go-build-toolchain:PR-25:16                            53 M
```

`golang` is installed from the PR.

It does have the problem however that the [Ubuntu build fails](https://jenkins-1.wolf.hpdd.intel.com:9090/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-go-build-toolchain-PR-25-master/32/pipeline/181/) because the _go-build-toolchain_ PR does not build for Ubuntu.

It's going to need to however, as Ubuntu has just become an officially supported distro for DAOS 2.8.

That said, this is an inherent problem with `CI_PR_REPOS`.  Unlike the `PR-repos[-<distro>]:` commit pragmas `CI_PR_REPOS` doesn't have per-distro control, so the implementation could use commit pragmas for the downstream test instead of setting the parameter.

But I think given that the _go-build-toolchain_ is about the only project that requires the build stage in the downstream job, we can punt on using pragmas instead of parameters and just have _go-build-toolchain_ build for Ubuntu also.

**Comment by brianjmurrell** [comment]
> Can we somehow split the work as Ubuntu is required for 2.8 but the new go-build-toolchain is required for 2.6?

The problem is that we have no mechanism to skip the Ubuntu build downstream when using `CI_PR_REPOS` and so every go-build-toolchain PR is going to fail it's downstream testing.

You could switch to using commit pragmas in the downstream testing (there is facility for that -- or almost at least perhaps) but given the simplicity of the packaging for the go-build-toolchain repo, a `debian/` dir for it would probably be pretty trivial.  There is a ticket already created specifically for that effort, FWIW, as it is going to be necessary sooner than later anyway.

**Review by JohnMalmberg — APPROVED** [review]

**Review by ryon-jensen — APPROVED** [review]

File: utils/rpms/packaging/Makefile_packaging.mk
Comment (brianjmurrell) [reply utils/rpms/packaging/Makefile_packaging.mk:1]: Could/should be.  But Jenkins.

Once we move this all to GitHub Actions, I am going to make it a submodule.  Or it might all just be part of an RPM-building GH Action.  But yes, having to use a "_manual_ submodule" here sucks.  Just another Jenkins issue.

**Pull request merged**
